### PR TITLE
Hit Test Redesign + new Input

### PR DESCRIPTION
**Pull request by mmgaw113**
**Sample**

Hit Test

**Summary**

- Add support for Enhanced Input System 
- Change key binding from Left Mouse Click to Left Shift + Left Mouse Click
- Change UI Instructions
- Add Standalone Blueprint that can be added into any scene.
- Update Readme

**Tests**

Windows

**ArcGIS Maps SDK Version**

Maps SDK 1.4 w/ UE 5.0

### CODE REVIEW

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 1]
```diff
@@ -23,12 +23,13 @@ Get the ID for individual buildings in a scene.
...
-3. Within the Level Blueprint, create an Input mapping. Off the **Pressed** node, drag off and create a **Line Trace by Channel**.
-4. Drag off the Exec node, and create an **ArcGISRaycast Hit**.
-5. Break the **ArcGISRaycast Hit Result** and drag off the **FeatureID** node.
-6. Convert this integer to a string, and connect it to a print node.
-7. Enter your **APIKey** in the **APIMapCreator**.
-8. Hit play. Once the scene is running click a building. The building's ID's will show up in a debug on screen. 
+3. Within the HitTest folder, drag the ArcGISRaycast Blueprint into the scene.
+4. In the UI folder within the HitTest folder, drag the bp_featureId actor into the scene.
+5. In the outliner, click on the ArcGISRaycast Actor.
+6. Add a Speher Static Mesh Actor into the scene and call it 'HitLocation'.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:29]:
```suggestion
6. In the Outliner panel, select the ArcGISRaycast Actor.
```

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 2]
```diff
@@ -23,12 +23,13 @@ Get the ID for individual buildings in a scene.
...
+7. For each of these variables, click on the drop down and add the respective actor as it's reference.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:30]:
```suggestion
7. Navigate to the Default sections in the Details panel. For each of listed variables, click the drop-down and add the respective Actor as its reference.
```

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 3]
```diff
@@ -23,12 +23,13 @@ Get the ID for individual buildings in a scene.
...
+8. Enter your **APIKey** in the **APIMapCreator**.
+9. Hit play. Once the scene is running hold shift and click a building. The building's ID's will show up in a debug on screen.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:32]:
```suggestion
9. Hit Play. Once the Level is running, hold Shift and click a building. The building's IDs will show up in a debug on screen. 
```

**Review by avr757 — CHANGES_REQUESTED** [review]
Some edits on the instructions:

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 4]
```diff
@@ -23,12 +23,13 @@ Get the ID for individual buildings in a scene.
...
-5. Break the **ArcGISRaycast Hit Result** and drag off the **FeatureID** node.
-6. Convert this integer to a string, and connect it to a print node.
-7. Enter your **APIKey** in the **APIMapCreator**.
-8. Hit play. Once the scene is running click a building. The building's ID's will show up in a debug on screen. 
+3. Within the HitTest folder, drag the ArcGISRaycast Blueprint into the scene.
+4. In the UI folder within the HitTest folder, drag the bp_featureId actor into the scene.
+5. In the outliner, click on the ArcGISRaycast Actor.
+6. Add a Speher Static Mesh Actor into the scene and call it 'HitLocation'.
+7. For each of these variables, click on the drop down and add the respective actor as it's reference.
+8. Enter your **APIKey** in the **APIMapCreator**.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:31]: Did you mean ArcGISMapComponent? Just wanted to make sure.

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 5]
```diff
@@ -1,38 +1,39 @@
...
-2. Add an **ArcGISPawn** to the scene.
-3. Within the Level Blueprint, create an Input mapping. Off the **Pressed** node, drag off and create a **Line Trace by Channel**.
-4. Drag off the Exec node, and create an **ArcGISRaycast Hit**.
-5. Break the **ArcGISRaycast Hit Result** and drag off the **FeatureID** node.
-6. Convert this integer to a string, and connect it to a print node.
-7. Enter your **APIKey** in the **APIMapCreator**.
-8. Hit play. Once the scene is running click a building. The building's ID's will show up in a debug on screen. 
+2. Add an **ArcGISPawn** to the Level.
+3. Within the HitTest folder, drag the ArcGISRaycast Blueprint into the Level.
+4. In the UI folder within the HitTest folder, drag the bp_featureId actor into the Level.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:27]:
```suggestion
4. In the UI folder within the HitTest folder, drag the bp_featureId Actor into the Level.
```
Sorry 1 more change, I changed the Actor to capitalized to align with Unreal term

**Comment by mmgaw113** [comment]
Forgot to upload an image to show the variables. So that is there now and changed some wording based on that.

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 6]
```diff
@@ -1,38 +1,42 @@
...
-3. Within the Level Blueprint, create an Input mapping. Off the **Pressed** node, drag off and create a **Line Trace by Channel**.
-4. Drag off the Exec node, and create an **ArcGISRaycast Hit**.
-5. Break the **ArcGISRaycast Hit Result** and drag off the **FeatureID** node.
-6. Convert this integer to a string, and connect it to a print node.
-7. Enter your **APIKey** in the **APIMapCreator**.
-8. Hit play. Once the scene is running click a building. The building's ID's will show up in a debug on screen. 
+2. Add an **ArcGISPawn** to the Level.
+3. Within the HitTest folder, drag the ArcGISRaycast Blueprint into the Level.
+4. In the UI folder within the HitTest folder, drag the bp_featureId Actor into the Level.
+5. Add a Speher Static Mesh Actor into the Level and call it 'HitLocation'.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:28]:
```suggestion
5. Add a Sphere Static Mesh Actor into the Level and call it 'HitLocation'.
```

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 7]
```diff
@@ -1,38 +1,42 @@
...
+6. In the outliner, click on the ArcGISRaycast Actor.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:29]:
```suggestion
6. In the Outliner panel, select the ArcGISRaycast Actor.
```

File: sample_project/Content/SampleViewer/Samples/HitTest/readme.md [hunk 8]
```diff
@@ -1,38 +1,42 @@
...
+7. Navigate to the Default sections in the Details Panel. For each variable shown in the image, click on the drop-down and add the respective Actor as its reference.
+
+![Image of Variables](RaycastComponent.png)
+
+8. Enter your **APIKey** in the **Map Component**.
```
Comment (avr757) [inline sample_project/Content/SampleViewer/Samples/HitTest/readme.md:34]:
```suggestion
8. Select the **ArcGISMap** Actor in the Outliner panel, and enter your **APIKey** in the Authentication section in the Details panel.
```

**Review by avr757 — APPROVED** [review]
Thanks for including a screenshot image!

**Pull request merged**